### ISSUE TITLE
IOS. Unity 5.6 + GVR 1.4 native GVR - setting DefaultDeviceProfile does not work

### ISSUE THREAD

**Issue by otri** [googlevr/gvr-unity-sdk#568]
I cannot get the DefaultDeviceProfile to stick Using release of GVR 1.4 on Unity 5.6.0f3, with VR enabled and Cardboard API added.

The app always initializes to using Default Cardboard.  This works fine in Unity 5.5.0f3 and GoogleVR 1.4, without native GVR integration.

![switch viewer - default cardboard](https://cloud.githubusercontent.com/assets/949250/25976747/be987864-366b-11e7-8482-6027ec8d823d.PNG)

Also... I would like to be able to override the Cardboard currently active Device Profile and hide the settings gear if I plug my iPhone into a known headset, so user does not have to do any steps.  Is there some way I can set the DeviceProfile at runtime?

![settings gear](https://cloud.githubusercontent.com/assets/949250/25976751/c31c14e0-366b-11e7-9246-6a8a0c84e0ef.PNG)

**Comment by miraleung** [comment]
Could you please reply with the number of the GVR SDK in the Podfile?

**Comment by otri** [comment]
It's certainly GVR SDK 1.40.

Updated GvrViewer.cs lines 381 to report the setter error:
```
    if (DefaultDeviceProfile != null) {
      if( device.SetDefaultDeviceProfile(DefaultDeviceProfile) == false ) {
        Debug.LogWarning("Failed to set DefaultDeviceProfile, on device:" + device.ToString());
      }
    }
```

I get a consistent warning:

> Failed to set DefaultDeviceProfile, on device:Gvr.Internal.iOSDevice

**Comment by miraleung** [comment]
GvrViewer.cs will be going away as of 1.50, and this codepath shouldn't be running for Unity 5.6.

**Comment by otri** [comment]
Okay, is there an alternative way for specifying a headset profile when plugging or attaching in some way to a known headset?  I'd like to eliminate user error if possible, and a guaranteed optical match to a physically attached headset would be best.

**Comment by miraleung** [comment]
The GVR Unity SDK exposes only the available methods from the GVR NDK / SDK; could you please try filling an issue in gvr-android-sdk?

**Comment by miraleung** [comment]
We are investigating internally on the Default profile initialization, thanks for the report.

**Comment by peterept** [comment]
Any update on this? Is there a work-around to load a different profile with Unity3d + GoogleVR SDK?

**Comment by otri** [comment]
Hoping there's some movement on this one.  We need to set a GoogleVR Default headset profile for our headset, or we'll have to stop supporting it completely.

**Comment by miraleung** [comment]
Yes, this method has been added in the last release; please take a look at Assets/GoogleVR/Scripts/GvrCardboardHelpers.cs.

**Comment by otri** [comment]
Thanks Mira!  Will give it a shot here shortly.

**Comment by guitarrapc** [comment]
Great tip! BTW, is there any way to check what DeviceProfile currently sat? I mean I do want to set default DeviceProfile only if there are no other device profiler has been sat before my app.

**Comment by otri** [comment]
Yeah, good point. I would open a separate issue.  These properties are important to know and override for streamlining the user experience with a specific piece of hardware.

I can confirm this issue is closed as of GoogleVR 1.70.
